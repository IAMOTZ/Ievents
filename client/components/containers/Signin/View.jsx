import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { RegularTopNavigation } from '../../common/TopNavigation';
import Footer from '../../common/Footer';
import { LoadingIcon } from '../../common/LoadingAnimation';
import { BigAlert, SmallAlert } from '../../common/Alert';

const View = props => (
  <div id="sign-in-container">
    <RegularTopNavigation />
    <div className="d-flex flex-column align-items-center page-content">
      <LoadingIcon start={props.loggingUserStarted} size={3} />
      <div className="card card-form">
        <h1 className="card-header">Log in</h1>
        <div className="card-body">
          <BigAlert message={props.loggingUserError ? 'Email or password incorrect' : null} />
          <form>
            <div className="form-group">
              <label htmlFor="email">Email</label>
              <div className="input-group">
                <div className="input-group-addon">
                  <i className="fa fa-envelope" />
                </div>
                <input
                  type="email"
                  className="form-control"
                  id="email"
                  placeholder="Your Email"
                  name="email"
                  onChange={props.getInput}
                />
              </div>
              <SmallAlert message={props.inputErrors.emailError} />
            </div>
            <div className="form-group">
              <label htmlFor="password">Password</label>
              <div className="input-group">
                <div className="input-group-addon">
                  <i className="fa fa-user-secret" />
                </div>
                <input
                  type="password"
                  className="form-control"
                  id="password"
                  placeholder="Password"
                  name="password"
                  onChange={props.getInput}
                />
              </div>
              <SmallAlert message={props.inputErrors.passwordError} />
            </div>
            <button
              className="btn btn-block ie-blue-button text-white"
              disabled={props.loggingUserStarted}
              onClick={props.login}
            >Log in
            </button>
            <div className="text-center mt-2" style={{ fontSize: '15px' }}>
              <span>{'Don\'t have an account?'}</span>&nbsp;
              <Link to="/users/signup">Sign up</Link>
            </div>
          </form>
        </div>
      </div>
    </div>
    <Footer />
  </div>
);

View.defaultProps = {
  loggingUserError: '',
  inputErrors: {},
};

/* eslint-disable react/forbid-prop-types */
View.propTypes = {
  loggingUserStarted: PropTypes.bool.isRequired,
  loggingUserError: PropTypes.string,
  inputErrors: PropTypes.object,
  getInput: PropTypes.func.isRequired,
  login: PropTypes.func.isRequired,
};

export default View;
