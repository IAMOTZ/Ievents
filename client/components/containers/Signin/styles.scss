$custom-black: rgba(0,0,0, 0.9);
$custom-blue: rgb(0, 142, 214);

#sign-in-container {
  .page-content {
    margin-top: 100px;
    min-height: 70vh;
  }
  .card-form {
    margin-bottom: 95px;
    width: 350px;
    .card-title {
      background-color: $custom-black;
      color: white;
    }
  }
}