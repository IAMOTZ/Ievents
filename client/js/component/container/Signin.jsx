import React from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';

import signupStyles from '../../../sass/signup.scss';
import TopNavigation from '../common/TopNavigation.jsx';
import Footer from '../common/Footer.jsx';
import { WarningAlert } from '../common/Alert';

import { loginUser, clearError } from '../../actions/authAction';

@connect((store) => {
  return {
    user: store.user,
    error: store.user.status.error.message,    
  }
})

export default class Signin extends React.Component {
  constructor() {
    super();
    this.state = {
      email: null,
      password: null,
    };
  }

  componentWillUnmount() {
    this.props.dispatch(clearError());
  }

  // This method uses user input to update the state
  getInput = (e) => {
    const state = this.state;
    state[e.target.name] = e.target.value;
    this.setState(state);
  }

  // This method fires the action to log in a user
  login = () => {
    const {
     email, password,
    } = this.state;
    this.props.dispatch(loginUser({
      email,
      password
    }));
  }

  render() {
    if (this.props.user.status.fetched === true) {
      return (
        <div>
          <Redirect to="/addEvent" />
        </div>
      );
    } else {
      return (
        <div>
          <TopNavigation />
          <div className="the-flex-box d-flex flex-column align-items-center">
            <div className="card card-form">
              <WarningAlert message={this.props.error}/>
              <h1 className="card-header">Sign in</h1>
              <div className="card-body">
                <form>
                  <div className="form-group">
                    <label for="email">Email</label>
                    <div className="input-group">
                      <div className="input-group-addon">
                        <i className="fa fa-envelope"></i>
                      </div>
                      <input type="email" className="form-control" id="email" placeholder="Your Email" name="email" onChange={this.getInput} />
                    </div>
                  </div>
                  <div className="form-group">
                    <label for="password">Password</label>
                    <div className="input-group">
                      <div className="input-group-addon">
                        <i className="fa fa-user-secret"></i>
                      </div>
                      <input type="password" className="form-control" id="password" placeholder="Password" name="password" onChange={this.getInput} />
                    </div>
                  </div>
                  <a className="btn btn-block dark-button text-white" onClick={this.login}>Log in</a>
                  <div className="text-center mt-2">
                    <a href="" className="text-muted">forgot password?</a>
                  </div>
                </form>
              </div>
            </div>

            <Footer />
          </div>

        </div>
      );
    }

  }
}
