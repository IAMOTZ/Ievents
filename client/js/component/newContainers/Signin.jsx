import React from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import { stopAsyncProcess } from '../../actions/commonActions';
import * as asyncProcess from '../../actions/asyncProcess';
import { validateSigninInputs } from '../../helpers/inputValidators';
import { loginUser } from '../../actions/authAction';
import TopNavigation from '../common/TopNavigation';
import Footer from '../common/Footer';
import { LoadingIcon } from '../common/LoadingAnimation';
import { BigAlert, SmallAlert } from '../common/Alert';

@connect(store => (
  {
    user: store.authReducer.user,
    loggingUserStarted: store.authReducer.loggingUserStarted,
    loggingUserResolved: store.authReducer.loggingUserResolved,
    loggingUserError: store.authReducer.loggingUserError,
  }
))
class Signin extends React.Component {
  constructor() {
    super();
    this.state = {
      email: null,
      password: null,
      inputErrors: {
        emailError: null,
        passwordError: null,
      },
    };
  }

  componentWillUnmount() {
    this.props.dispatch(stopAsyncProcess(asyncProcess.LOGGING_USER));
  }

  /**
   * Stores the user inputs in the state of this component.
   * @param {Event} event The event object.
   */
  getInput = (event) => {
    const state = { ...this.state };
    state[event.target.name] = event.target.value;
    this.setState(state);
  }

  /**
   * Dispatches an action to authenticate a user.
   */
  login = (event) => {
    event.preventDefault();
    this.props.dispatch(stopAsyncProcess(asyncProcess.LOGGING_USER));
    const { email, password } = this.state;
    const inputErrors = validateSigninInputs(this.state);
    if (inputErrors.errorFound) {
      const state = { ...this.state };
      state.inputErrors = inputErrors;
      this.setState(state);
    } else {
      this.clearInputErrors();
      this.props.dispatch(loginUser({ email, password }));
    }
  }

  clearInputErrors = () => {
    const state = { ...this.state };
    state.inputErrors = {
      emailError: null,
      passwordError: null,
    };
    this.setState(state);
  }

  render() {
    if (this.props.loggingUserResolved) {
      return (<Redirect to="/addEvent" />);
    }
    return (
      <div id="sign-in-container">
        <TopNavigation />
        <div className="d-flex flex-column align-items-center main-content">
          <LoadingIcon start={this.props.loggingUserStarted} size={3} />
          <div className="card card-form">
            <h1 className="card-header">Sign in</h1>
            <div className="card-body">
              <BigAlert message={this.props.loggingUserError ? 'email or password incorrect' : null} />
              <form>
                <div className="form-group">
                  <label htmlFor="email">Email</label>
                  <div className="input-group">
                    <div className="input-group-addon">
                      <i className="fa fa-envelope" />
                    </div>
                    <input
                      type="email"
                      className="form-control"
                      id="email"
                      placeholder="Your Email"
                      name="email"
                      onChange={this.getInput}
                    />
                  </div>
                  <SmallAlert message={this.state.inputErrors.emailError} />
                </div>
                <div className="form-group">
                  <label htmlFor="password">Password</label>
                  <div className="input-group">
                    <div className="input-group-addon">
                      <i className="fa fa-user-secret" />
                    </div>
                    <input
                      type="password"
                      className="form-control"
                      id="password"
                      placeholder="Password"
                      name="password"
                      onChange={this.getInput}
                    />
                  </div>
                  <SmallAlert message={this.state.inputErrors.passwordError} />
                </div>
                <button
                  className="btn btn-block dark-button text-white"
                  disabled={this.props.loggingUserStarted}
                  onClick={this.login}
                >Log in
                  </button>
                <div className="text-center mt-2">
                  <a href="" className="text-muted">forgot password?</a>
                </div>
              </form>
            </div>
          </div>
          <Footer />
        </div>
      </div>
    );
  }
}

export default Signin;
