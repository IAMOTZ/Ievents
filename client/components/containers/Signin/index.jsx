import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { validateSigninInputs } from '../../../helpers/inputValidators';
import { loginUser } from '../../../actions/authAction';
import { stopAsyncProcess } from '../../../actions/commonActions';
import * as asyncProcess from '../../../actions/asyncProcess';
import './styles.scss';
import View from './View';

@connect(store => (
  {
    loggingUserStarted: store.authReducer.loggingUserStarted,
    loggingUserResolved: store.authReducer.loggingUserResolved,
    loggingUserError: store.authReducer.loggingUserError,
  }
))
class Signin extends React.Component {
  constructor() {
    super();
    this.state = {
      email: null,
      password: null,
      inputErrors: {
        emailError: null,
        passwordError: null,
      },
    };
  }

  componentWillUnmount() {
    this.props.dispatch(stopAsyncProcess(asyncProcess.LOGGING_USER));
  }

  /**
   * Stores the user inputs in the state of this component.
   * @param {Event} event The event object.
   */
  getInput = (event) => {
    const state = { ...this.state };
    state[event.target.name] = event.target.value;
    this.setState(state);
  }

  /**
   * Dispatches an action to authenticate a user.
   */
  login = (event) => {
    event.preventDefault();
    this.props.dispatch(stopAsyncProcess(asyncProcess.LOGGING_USER));
    const { email, password } = this.state;
    const inputErrors = validateSigninInputs(this.state);
    if (inputErrors.errorFound) {
      const state = { ...this.state };
      state.inputErrors = inputErrors;
      this.setState(state);
    } else {
      this.clearInputErrors();
      this.props.dispatch(loginUser({ email, password }));
    }
  }

  clearInputErrors = () => {
    const state = { ...this.state };
    state.inputErrors = {
      emailError: null,
      passwordError: null,
    };
    this.setState(state);
  }

  render() {
    if (this.props.loggingUserResolved) {
      return (<Redirect to="/events" />);
    }
    return (
      <View
        loggingUserStarted={this.props.loggingUserStarted}
        loggingUserError={this.props.loggingUserError}
        inputErrors={this.state.inputErrors}
        getInput={this.getInput}
        login={this.login}
      />);
  }
}


Signin.defaultProps = {
  loggingUserStarted: false,
  loggingUserResolved: false,
  loggingUserError: '',
  dispatch: () => {},
};

Signin.propTypes = {
  loggingUserStarted: PropTypes.bool,
  loggingUserResolved: PropTypes.bool,
  loggingUserError: PropTypes.string,
  dispatch: PropTypes.func,
};

export default Signin;
