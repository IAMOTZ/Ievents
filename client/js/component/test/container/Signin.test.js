/* global shallow */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import Signin from '../../container/Signin.jsx';

const props = {
  user: {},
  error: null,
  status: {
    fetching: false,
    fetched: false,
  },
  dispatch: () => { },
};

const alterPorps = newProps => ({
  user: {},
  error: newProps.error || null,
  status: {
    fetching: newProps.fetching || false,
    fetched: newProps.fetched || false,
  },
  dispatch: () => { },
});

describe('<Signin />', () => {
  describe('Rendering:', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<Signin {...props} />);
    })
    it('should render correctly on page load', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('should redirect if user is already logged in', () => {
      const newProps = alterPorps({ fetched: true });
      wrapper.setProps(newProps);
      expect(wrapper).toMatchSnapshot();
    });
    it('should render correctly when logging in a user', () => {
      const newProps = alterPorps({ fetching: true });
      wrapper.setProps(newProps);
      expect(wrapper).toMatchSnapshot();
    });
    it('should render correctly when there is an error', () => {
      const newProps = alterPorps({ error: 'there was an error' });
      wrapper.setProps(newProps);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Behaviour:', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<Signin {...props} />);
    });
    it('should update the state as the user provides email input', () => {
      const email = 'test@gmail.com';
      wrapper
        .find('#email')
        .simulate('change', { target: { name: 'email', value: email } });
      expect(wrapper.state('email')).toEqual(email);
    });
    it('should update the state as the user provides password input', () => {
      const password = 'Test-Password12';
      wrapper
        .find('#password')
        .simulate('change', { target: { name: 'password', value: password } });
      expect(wrapper.state('password')).toEqual(password);
    });
  });
});
