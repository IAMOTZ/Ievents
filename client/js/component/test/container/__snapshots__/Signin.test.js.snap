// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signin /> Rendering: should redirect if user is already logged in 1`] = `
<Redirect
  push={false}
  to="/addEvent"
/>
`;

exports[`<Signin /> Rendering: should render correctly on page load 1`] = `
<div
  id="sign-in-container"
>
  <TopNav />
  <div
    className="d-flex flex-column align-items-center main-content"
  >
    <LoadingIcon
      size={3}
      start={false}
    />
    <div
      className="m-2"
    >
      <Component
        message={null}
      />
    </div>
    <div
      className="card card-form"
    >
      <h1
        className="card-header"
      >
        Sign in
      </h1>
      <div
        className="card-body"
      >
        <form>
          <div
            className="form-group"
          >
            <label
              htmlFor="email"
            >
              Email
            </label>
            <div
              className="input-group"
            >
              <div
                className="input-group-addon"
              >
                <i
                  className="fa fa-envelope"
                />
              </div>
              <input
                className="form-control"
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Your Email"
                type="email"
              />
            </div>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="password"
            >
              Password
            </label>
            <div
              className="input-group"
            >
              <div
                className="input-group-addon"
              >
                <i
                  className="fa fa-user-secret"
                />
              </div>
              <input
                className="form-control"
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
              />
            </div>
          </div>
          <button
            className="btn btn-block dark-button text-white"
            disabled={false}
            onClick={[Function]}
          >
            Log in
          </button>
          <div
            className="text-center mt-2"
          >
            <a
              className="text-muted"
              href=""
            >
              forgot password?
            </a>
          </div>
        </form>
      </div>
    </div>
    <Footer />
  </div>
</div>
`;

exports[`<Signin /> Rendering: should render correctly when logging in a user 1`] = `
<div
  id="sign-in-container"
>
  <TopNav />
  <div
    className="d-flex flex-column align-items-center main-content"
  >
    <LoadingIcon
      size={3}
      start={true}
    />
    <div
      className="m-2"
    >
      <Component
        message={null}
      />
    </div>
    <div
      className="card card-form"
    >
      <h1
        className="card-header"
      >
        Sign in
      </h1>
      <div
        className="card-body"
      >
        <form>
          <div
            className="form-group"
          >
            <label
              htmlFor="email"
            >
              Email
            </label>
            <div
              className="input-group"
            >
              <div
                className="input-group-addon"
              >
                <i
                  className="fa fa-envelope"
                />
              </div>
              <input
                className="form-control"
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Your Email"
                type="email"
              />
            </div>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="password"
            >
              Password
            </label>
            <div
              className="input-group"
            >
              <div
                className="input-group-addon"
              >
                <i
                  className="fa fa-user-secret"
                />
              </div>
              <input
                className="form-control"
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
              />
            </div>
          </div>
          <button
            className="btn btn-block dark-button text-white"
            disabled={true}
            onClick={[Function]}
          >
            Log in
          </button>
          <div
            className="text-center mt-2"
          >
            <a
              className="text-muted"
              href=""
            >
              forgot password?
            </a>
          </div>
        </form>
      </div>
    </div>
    <Footer />
  </div>
</div>
`;

exports[`<Signin /> Rendering: should render correctly when there is an error 1`] = `
<div
  id="sign-in-container"
>
  <TopNav />
  <div
    className="d-flex flex-column align-items-center main-content"
  >
    <LoadingIcon
      size={3}
      start={false}
    />
    <div
      className="m-2"
    >
      <Component
        message="there was an error"
      />
    </div>
    <div
      className="card card-form"
    >
      <h1
        className="card-header"
      >
        Sign in
      </h1>
      <div
        className="card-body"
      >
        <form>
          <div
            className="form-group"
          >
            <label
              htmlFor="email"
            >
              Email
            </label>
            <div
              className="input-group"
            >
              <div
                className="input-group-addon"
              >
                <i
                  className="fa fa-envelope"
                />
              </div>
              <input
                className="form-control"
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Your Email"
                type="email"
              />
            </div>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="password"
            >
              Password
            </label>
            <div
              className="input-group"
            >
              <div
                className="input-group-addon"
              >
                <i
                  className="fa fa-user-secret"
                />
              </div>
              <input
                className="form-control"
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
              />
            </div>
          </div>
          <button
            className="btn btn-block dark-button text-white"
            disabled={false}
            onClick={[Function]}
          >
            Log in
          </button>
          <div
            className="text-center mt-2"
          >
            <a
              className="text-muted"
              href=""
            >
              forgot password?
            </a>
          </div>
        </form>
      </div>
    </div>
    <Footer />
  </div>
</div>
`;
